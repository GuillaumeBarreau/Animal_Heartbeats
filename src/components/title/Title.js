import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

const TitleApp = styled.h1`
	color: ${props => props.theme.text};
    margin: 0;
    padding: 2rem 0 0;
`;

const Title = ({ children, theme }) => <TitleApp theme={theme}>{children}</TitleApp>;

const mapStateToProps = state => {
	return {
		theme: state.theme,
	};
};

export default connect(mapStateToProps)(Title);