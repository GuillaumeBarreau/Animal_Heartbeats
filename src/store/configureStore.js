import { createStore, applyMiddleware } from 'redux';
import reducers from './reducers/combineReducers';

function logger({ getState }) {
	return next => action => {
		console.log('will dispatch', action);
		return next(action);
	};
}

export const store = createStore(reducers, applyMiddleware(logger));
 