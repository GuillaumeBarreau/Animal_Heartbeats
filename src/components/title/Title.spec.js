import { shallow } from 'enzyme';
import React from 'react';
import Title from './Title';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('Title component', () => {
	let children;
	let store;

	beforeEach(() => {
		children = 'children';
		store = mockStore({
			theme: 'dark',
		});
	});

	it('Render Title component', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Title>{children}</Title>
			</Provider>
		);
		expect(cmp.exists()).toBe(true);
	});

	it('Snapshot testing', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Title>{children}</Title>
			</Provider>
		);
		expect(cmp.html()).toMatchSnapshot();
	});

	it('should display childrens', () => {
		const cmp = shallow(
			<Provider store={store}>
				<Title>{children}</Title>
			</Provider>
		);
		expect(cmp.html()).toContain('children');
	});
});
