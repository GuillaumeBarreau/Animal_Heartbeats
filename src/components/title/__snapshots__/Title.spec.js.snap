// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Snapshot testing 1`] = `"<h1 class=\\"sc-AykKC fdPqkt\\">children</h1>"`;
